packages/rest-explorer/src/rest-explorer.controller.ts

```diff
@@ -66,7 +87,17 @@ export class ExplorerController {
       .send(homePage);
   }
 
+  spec() {
+    const specObj = this.restServer.getApiSpec(this.requestContext);
+    const spec = JSON.stringify(specObj, null, 2);
+    this.response.setHeader('content-type', 'application/json; charset=utf-8');
+    this.response.end(spec, 'utf-8');
```

Comment:
Is it necessary for this endpoint to deal with JSON serialization to HTTP response? I believe that by default, LoopBack is automatically serializing controller return values as JSON. I think this controller method can be simplified as follows:

```ts
spec() {
  return this.restServer.getApiSpec(this.requestContext);
}
```

Reply:
You may well be right.  This code is copy-pasta from `RestServer._serveOpenApiSpec`.

Reply:
Doing some git history spelunking, it looks like this code comes from back when it was done at the express level, instead of being a controller level (3286bc65).

A quick local test indicates that your simplification does work, so I'll clean up both copies of this code :+1:

Reply:
One thing I notice happens when I do this: the response is no longer _explicitly_ listed (in the HTTP headers) as being in the UTF-8 encoding (no specific encoding is listed).

I suspect this is largely a non-issue, but let me know if you disagree.

Reply:
> One thing I notice happens when I do this: the response is no longer explicitly listed (in the HTTP headers) as being in the UTF-8 encoding (no specific encoding is listed).

That's because of the following line:

https://github.com/strongloop/loopback-next/blob/c944654881140119e44fe9e1db85a0f26e59d658/packages/rest/src/writer.ts#L52

I think the missing charset information is fine as far as this pull request is concerned.

It would be nice to eventually fix the line above to include charset/encoding too - would you like to open a new pull request?